app/scripts/lib/transaction/util.ts

```diff
@@ -260,19 +262,19 @@ function validateSecurity(request: AddTransactionRequest) {
 
     const securityAlertId = generateSecurityAlertId();
 
-    validateRequestWithPPOM({
+    const securityAlertResponse = await validateRequestWithPPOM({
```

Comment:
This is all intentionally not awaited as we don't want to block the confirmation while the validation happens.

Hence why we immediately set the loading response, but handle any subsequent actual response with a `then`.

Maybe worth a comment to clarify that for future.

Reply:
Sorry, I reverted it and added a comment to avoid that happening again.